src/bootstrap/build.rs

```diff
@@ -1,18 +1,33 @@
+use env::consts::{EXE_EXTENSION, EXE_SUFFIX};
 use std::env;
+use std::ffi::OsString;
 use std::path::PathBuf;
 
+/// Given an executable called `name`, return the filename for the
+/// executable for a particular target.
+pub fn exe(name: &PathBuf) -> PathBuf {
+    if EXE_EXTENSION != "" && name.extension() != Some(EXE_EXTENSION.as_ref()) {
```

Comment:
It seems like this method is basically `name.with_extension(EXE_EXTENSION)`? Unless I'm missing something, that's the intent of the code...

Reply:
The difference is this doesn't replace any existing extension. with_extension will change `my.file` to `my.exe`.